crates/bitwarden-vault/src/cipher/cipher_client/delete_attachment.rs

```diff
@@ -0,0 +1,453 @@
+use bitwarden_api_api::{apis::ciphers_api::CiphersApi, models};
+use bitwarden_core::require;
+use bitwarden_state::repository::Repository;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::wasm_bindgen;
+
+use crate::{Cipher, CipherError, CipherId, CiphersClient, VaultParseError};
+
+/// Standalone function to delete an attachment from a cipher that is extracted for ease of unit
+/// testing.
+async fn delete_attachment(
+    api_client: &dyn CiphersApi,
+    repository: &dyn Repository<Cipher>,
+    cipher_id: CipherId,
+    attachment_id: &str,
+    admin: bool,
+) -> Result<Cipher, CipherError> {
+    let response = if admin {
+        api_client
+            .delete_attachment_admin(cipher_id.into(), attachment_id)
+            .await?
+    } else {
+        api_client
+            .delete_attachment(cipher_id.into(), attachment_id)
+            .await?
+    };
+
+    let boxed_cipher: Box<models::Cipher> = require!(response.cipher);
+    let cipher_response = *boxed_cipher;
+    let mut cipher = require!(repository.get(cipher_id.to_string()).await?);
+
+    cipher.revision_date = require!(cipher_response.revision_date)
+        .parse()
+        .map_err(Into::<VaultParseError>::into)?;
+
+    if let Some(ref mut attachments) = cipher.attachments {
+        attachments.retain(|a| a.id.as_deref() != Some(attachment_id));
+    }
+
+    repository
+        .set(cipher_id.to_string(), cipher.clone())
+        .await?;
+
+    Ok(cipher)
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+impl CiphersClient {
+    /// Delete an attachment from a cipher
+    pub async fn delete_attachment(
+        &self,
+        cipher_id: CipherId,
+        attachment_id: String,
```

Comment:
We can change the attachment id parameters for these two functions to be `&str`, no need to consume a string when a reference to it would be enough